lib/middleware/prototype-admin-routes.js

```diff
@@ -34,4 +34,20 @@ router.post('/password', (req, res) => {
   }
 });
 
+router.get('/reset', (req, res) => {
+
+  const backHref = req.get('Referrer') || '/'
+  res.render('reset', {
+    backHref
+  });
+});
+
+router.all('/reset-session-data', (req, res) => {
+  req.session.data = {};
+  const returnPath = req.body.previousPath || '/'
+  res.render('reset-done', {
+    returnPath
+  });
```

Comment:
The problem with `res.render()` here is we're showing it after a POST

The watch task will reload the page (back to a GET) as soon as you edit a stylesheet etc

Can we redirect to a stable success page instead?

```suggestion
  const successPath = '/prototype-admin/reset-session-data-success'
  res.redirect(`${successPath}?returnPath=${returnPath}`);
```

Reply:
I think the issue with that would be that you could then visit that stable success page using a regular GET without having actually cleared the data first, which would be misleading.

I’d usually use the POST->redirect->GET pattern, but maybe this is an exception? The GOVUK prototype kit renders the success page from the POST too... 🤔